src/modules/sync/advancedDiff.ts

```diff
@@ -0,0 +1,720 @@
+import {
+  ContentTypeElements,
+  ContentTypeModels,
+  ContentTypeSnippetModels,
+  TaxonomyModels,
+} from "@kontent-ai/management-sdk";
+import chalk from "chalk";
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from "fs";
+import open from "open";
+import { dirname, resolve } from "path";
+
+import { DiffParams } from "../../commands/diff/diff.js";
+import { logError, logInfo, LogOptions } from "../../log.js";
+import {
+  isCountLimitation,
+  isDefaultValue,
+  isDependsOn,
+  isExternalIdReference,
+  isExternalIdReferenceArray,
+  isMaximumTextLength,
+  isObjectReference,
+  isObjectReferenceArray,
+  isValidationRegex,
+} from "../../utils/typeguards.js";
+import { RequiredCodename } from "../../utils/types.js";
+import { DiffModel, DiffObject } from "./types/diffModel.js";
+import {
+  AddIntoPatchOperation,
+  MovePatchOperation,
+  PatchOperation,
+  RemovePatchOperation,
+  ReplacePatchOperation,
+  ReplacePatchOperationValue,
+} from "./types/patchOperation.js";
+
+type DiffData = DiffModel & Omit<DiffParams, "targetApiKey" | "sourceApiKey">;
+type EntityPathRenderer = {
+  regex: RegExp;
+  render: (() => string) | ((match: string[]) => string);
+};
+type Operation = PatchOperation["op"];
+type TypeOrSnippet = ContentTypeModels.IAddContentTypeData | ContentTypeSnippetModels.IAddContentTypeSnippetData;
+type ElementOrTerm = ContentTypeElements.Element | TaxonomyModels.IAddTaxonomyRequestModel;
+type RenderFunction<T extends PatchOperation> = (patchOp: T) => string;
+type EntityType = "taxonomies" | "types" | "snippets";
+type EntityActionType = "added" | "updated" | "deleted";
+
+// for testing purposes, mocked date passed to spawned process in runCommand()
+const myDate = process.env.MOCKED_DATE ? new Date(process.env.MOCKED_DATE) : new Date();
+const dateGenerated = myDate.toUTCString();
+
+export const generateDiff = (diffData: DiffData) => {
```

Comment:
I would modularize this file a bit.

In this file I would keep only generateDiff function with probably some supporting functions, which will combine logic from other files - fileUtils.ts and htmlUtils.ts

Then i would create fileUtils.ts, which would have logic for creating folders, files and reading files content

Lastly I would create htmlUtils.ts, where there would be logic for resolving and rendering HTML

Whay do you think?

Reply:
I agree, it's getting kinda humongous to be in one file. I'll couple the methods accordingly and split them into separate files